backend-api/src/functions/adapters/dynamoDbAdapter.ts

```diff
@@ -32,10 +55,83 @@ export class DynamoDbAdapter {
     }),
   });
 
-  constructor(tableName: string) {
+  constructor(
+    tableName: string,
+    private readonly getItemToSessionConvertor = getDynamoDBItemToSessionConvertor,
+  ) {
     this.tableName = tableName;
   }
 
+  async getSessionWithState(
```

Comment:
I've tried to play with a mapped type for this but probably hit similar pitfalls to what you've tried historically! If it works, it works.